src/networkManager.ts

```diff
@@ -32,15 +32,35 @@ export class NetworkManager implements INetworkManager {
      */
     public startSession(): Promise<ISessionInfo> {
 
+        let _sessionInfo: ISessionInfo;
         return this._sessionManager.startSession()
             .then((sessionInfo) => {
-                return Promise.all([sessionInfo, this._webSocketManager.connect()]);
+                _sessionInfo = sessionInfo;
+                return this._webSocketManager.connect();
             })
-            .then(([sessionInfo, connected]) => {
-                if (!connected) {
+            .then(connected => {
+                if (connected) {
+                    return _sessionInfo;
+                } else {
                     console.error("Failed to connect web socket");
+
+                    // Is the session invalid even though it hadn't expired ? 
+                    //  - perhaps the auth settings have changes since the token was issued ?
+                    return this._sessionManager.requestSession()
+                        .then(session => {
+                            // all good, websocket connection failure was just a blip and will automatically reconnect ...
+                            return _sessionInfo;
+                        })
+                        .catch(error2 => {
```

Comment:
`error2`?

Reply:
Ahh, it was prevent a shadowed variable tslint error (that isn't there anymore anyway )